models/company/Company.js

```diff
@@ -78,4 +82,94 @@ CompanySchema.statics.findCompanyById = function (id, callback) {
   });
 }
 
+CompanySchema.statics.getCompanyByName = function(company_name, callback){
+  const Company = this;
+
+  if(!company_name || typeof company_name != 'string')
+    return callback('bad_request');
+
+  Company
+    .find({company_name: `${company_name}`})
+    .then(company => callback(null,company))
+    .catch(err => callback("database_error "+err))
+
+}
+
+CompanySchema.statics.findCompaniesByFilter = function(_filters, _options, callback) {
+  // findCompaniesByFilter returns a tuple of objects with Companies, Filters and Options or an error if it exists
+  // _filters: company_name (string), email (string)
+  // _options: limit(int), skip(int)
+
+  const Company = this;
+
+  const filters = {
+    $and: []
+  }, filter_values = [], options = {
+    limit: _options.limit && !isNan(parseInt(_options.limit)) ? parseInt(_options.limit) : 100,
+    skip:  _options.skip && !isNaN(parseInt(_options.skip)) ? parseInt(_options.skip) : 0
+  };
+
+  if(_filters && _filters.company_name && typeof _filters.company_name == 'string'){
+    filters.$and.push({
+      company_name: {
+        $regex: _filters.company_name.trim(),
+        $options: 'i'
+      }
+    })
+    filter_values.company_name = _filters.company_name.trim();
+  }
+
+  if(_filters && _filters.email && typeof _filters.email == 'string'){
+    filters.$and.push({
+      email: {
+        $regex: _filters.email.trim(),
+        $options: 'i'
+      }
+    })
+    filter_values.email = _filters.email.trim();
+  }
+
+  Company
+    .find(filters.$and.length ? filters: {})
+    .sort({ company_name: 1})
+    .skip(options.skip)
+    .limit(options.limit)
+    .then(companies => callback(null, {
+      companies,
+      filters: filter_values,
+      options
+    }))
+    .catch(err => callback('database_error'))
+}
+
+CompanySchema.statics.updateCompany = function (id, data, callback) {
+  const Company = this;
+
+
+  if(!id || !validator.isMongoId(id.toString()) || (data.password && data.password != "" && data.password.length < 6))
+    return callback('bad_request')
+
+  Company.findById(mongoose.Types.ObjectId(id.toString()), (err, company) =>{
+    if(err || !company) return callback('document_not_found');
+    if(typeof data.credit_amount == "number"){
+
+      Company.findByIdAndUpdate(mongoose.Types.ObjectId(id.toString()),{$set: {
```

Comment:
credit i password u update'lediğin gibi update leyebilirsin zaten o hareketi yapıyorsun kod daha kısa olur. Yorum yaz fonksiyona data'da olan ve update'lenebilen field'lar için. Password update'lemek opsiyonel olsa bence hoş olur ama sen bilirsin. Onun dışında yine noktalı virgüllere dikkat

Reply:
Burayı anlamadım, 

company.credit = data.credit_amount diyerek mi updatele demeye çalışıyorsun? Eğer dediğin şey buysa, bu kod database e değişikliği geri yazmıyor.